File: sdk/typescript/conformance.mjs
import fs from 'node:fs';
import { canonicalize, conformanceResult } from './qsol_fed_sdk.mjs';

const fixture = JSON.parse(fs.readFileSync(new URL('../../fixtures/phase6/conformance.json', import.meta.url), 'utf8'));
const result = conformanceResult(fixture);
process.stdout.write(canonicalize(result).toString('utf8') + '\n');

File: sdk/python/conformance.py
#!/usr/bin/env python3
from __future__ import annotations

import json
from pathlib import Path
import sys

HERE = Path(__file__).resolve().parent
ROOT = HERE.parents[1]
sys.path.insert(0, str(HERE))

from qsol_fed_sdk import canonicalize, conformance_result


def main() -> None:
    fixture = json.loads((ROOT / "fixtures/phase6/conformance.json").read_text(encoding="utf-8"))
    result = conformance_result(fixture)
    sys.stdout.buffer.write(canonicalize(result) + b"\n")


if __name__ == "__main__":
    main()
